analyzers/ruby/ruby_test.go

```diff
@@ -43,3 +45,48 @@ func TestCustomGemfileLockPath(t *testing.T) {
 
 	assert.NotEqual(t, analyzed, analyzedCustom)
 }
+
+func TestFallbackOnMissingBundler(t *testing.T) {
+	buildTarget := "github.com/fossas/fossa-cli/cmd/fossa"
+	useLockfileOptions := map[string]interface{}{
+		"strategy":          "lockfile",
+		"gemfile-lock-path": filepath.Join("testdata", "Gemfile.lock"),
+	}
+
+	useBundlerWithLockfilePathOptions := map[string]interface{}{
+		"strategy":          "list",
+		"gemfile-lock-path": filepath.Join("testdata", "Gemfile.lock"),
+	}
+
+	gemModuleUsingLockfile := module.Module{
+		Name:        "test",
+		Type:        pkg.Ruby,
+		BuildTarget: buildTarget,
+		Options:     useLockfileOptions,
+	}
+
+	gemModuleUsingFallbackToLockfile := module.Module{
+		Name:        "test",
+		Type:        pkg.Ruby,
+		BuildTarget: buildTarget,
+		Options:     useBundlerWithLockfilePathOptions,
+	}
+
+	lockfileBasedAnalyzer, lockfileBasedAnalyzerErr := ruby.New(gemModuleUsingLockfile)
+	fallbackBasedAnalyzer, fallbackBasedAnalyzerErr := ruby.New(gemModuleUsingFallbackToLockfile)
+
+	assert.NoError(t, lockfileBasedAnalyzerErr)
```

Comment:
You should `assert.NoError` directly after the call. The tests will fail faster.